src/TextField/TextField.js

```diff
@@ -73,6 +76,15 @@ export default class TextField extends Component {
      * The CSS class name of the root element.
      */
     className: PropTypes.string,
+    /**
+     * Whether the label should be displayed in an error state
+     */
+    error: PropTypes.bool,
+    /**
+     * Whether this label should indicate that the input
+     * is required.
+     */
+    required: PropTypes.bool,
```

Comment:
Why moving the back the `required ` property at the top level?
Isn't it better to scope our properties?